### ISSUE TITLE
[🐛] Snyk should ignore workspace packages when validating dependencies in yarn.lock

### ISSUE THREAD

**Issue by mtlewis** [snyk/snyk#2126]
- `node -v`: 14.6.0
- `npm -v`: 6.14.11
- `snyk -v`: 1.669.0
- OS: OSX
- Command run: `snyk monitor --yarn-workspaces --strict-out-of-sync=false`

### Expected behaviour

Packages present in the monorepo that appear in dependency listings in yarn.lock should be ignored.

### Actual behaviour

Packages in a monorepo using yarn workspaces can appear in dependency listings in the yarn.lock for that monorepo without having their own top-level entry. This occurs when packages inside the monorepo depend on non-monorepo packages, that themselves depend on other monorepo packages. Snyk currently interprets this situation as a problem with yarn.lock and fails with an error:

```
Monitoring <repository root>...

Dependency <monorepo package name>@<version> was not found in yarn.lock. Your package.json and yarn.lock are probably out of sync. Please run "yarn install" and try again.
```

### Steps to reproduce

Run `snyk monitor --yarn-workspaces --strict-out-of-sync=false` in a monorepo with the characteristic described above. The repo backstage/backstage is a concrete example of this: there's a package [example-app](https://github.com/backstage/backstage/blob/master/packages/app/package.json) which depends on [@roadiehq/backstage-plugin-travis-ci](https://github.com/RoadieHQ/backstage-plugin-travis-ci) which is not part of the monorepo, but itself depends on multiple packages which are.

**Comment by lili2311** [comment]
Thanks @mtlewis this has come up a few times, I've forwarded this to the relevant team. We have a few options here, either to default to the setting in this case or a better option to detect and gracefully handle "local" workspace packages which will not be present in the lockfile.

**Comment by mtlewis** [comment]
> ... default to the setting in this case ...

Hey @lili2311! Thanks for your response. Which setting are you referring to here? We're currently blocked adopting Snyk in Backstage by this issue, and I'm not aware of any setting we can use to work around it, so very keen to know if there is in fact a workaround!

**Comment by lili2311** [comment]
hi @mtlewis I took another look with the provided examples and unfortunately indeed there is currently no workaround. Normally skipping out of sync check is enough, however in the provided example the "missing" dependencies are not only "top level" dependencies but also the transitive dependencies as both can be pointing to a local workspace. 

I've raised a proposed change https://github.com/snyk/nodejs-lockfile-parser/pull/119 and speaking to the team regarding this, this would still be a temporary workaround and not a full feature support for local dependencies which is what ideally the team would like to introduce here. The actual feature to understand what is a local dependency and only skip those would take a lot more time and planning than my proposed workaround. You can 👀 on the PR for any discussions & I will share any significant updates here

**Comment by mtlewis** [comment]
Out of interest, is it possible to disable strict mode when importing projects via the Snyk web UI, or is this flag only supported in the CLI?

**Comment by lili2311** [comment]
@mtlewis the same setting is available in the UI under Language Settings. We had an internal chat with the team and we agreed to proceed with the proposed change, so I will address some of the comments and will let you know once it is released.

**Comment by mtlewis** [comment]
Amazing, thank you!

**Comment by lili2311** [comment]
hi @mtlewis just a quick update that I'm addressing all the comments today and will be getting the team to review. Had to pause for a few weeks as we are recording things for SnykCon.

**Comment by mtlewis** [comment]
Great stuff - thanks for the update @lili2311, we're excited to have another go integrating Snyk once this change lands!

**Comment by lili2311** [comment]
Hi @mtlewis please could you try again with the latest version of the CLI https://github.com/snyk/snyk/releases/tag/v1.696.0?

**Comment by mtlewis** [comment]
Looks to be working, thanks again @lili2311!
